Feature/milestone 4

NewIncome opened the pull request:
Add the first logic:
- add logic dir
- add calculate.js
- add operate.js
Implement both modules

NewIncome committed:
Add logic folder with 2 files to implement de logic

- add the calculate and operate js files

NewIncome committed:
Implement a basic version of the calculate.js module

- add the buttonName '+/-' implementation

NewIncome committed:
Implement a basic version of the operate module

- uses the big.js module
- it calculates the 4 basic operations

NewIncome committed:
Fix the calculate method to work with % and import the Calculate
in the App.js

bellom commented on src/components/App.js:
```diff
@@ -1,6 +1,8 @@
 import '../App.css';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+// eslint-disable-next-line no-unused-vars
```
### [Optional]
It would be nice if you could solve the codebase instead of making use of Eslint inline comment.
If you are using VSCode, you can delete the inline comment and hover over the error(configure eslint checks locally). And you will see your options. Click on the more info button that will take you to a GitHub repo where it is explained why you are getting this error. I find this feature very educational. Please, try to learn and implement the necessary rules so you don't have to keep this disabling line on top of each file. I believe you can do this! 💪😇

bellom commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,13 @@
+    total *= -1;
+    next *= -1;
+  }
+  if (next && buttonName === '%') next /= 100;
+  if (total && buttonName === '%') total /= 100;
```
According to the project requirement, we need to make use of the module;
![image](https://user-images.githubusercontent.com/31897434/100809444-6b6b3b00-3436-11eb-9612-673806acef9c.png)

Kindly add the module from Big.js in `src/logic/operate.js` instead of hard-coding.

### Heads-Up:
When it comes to `%` we only need one operand.
example: click `10` then `%` = 0.01

bellom commented on src/logic/operate.js:
```diff
@@ -0,0 +1,12 @@
+export default function operate(numOne, numTwo, operation) {
+  const x = Big(numOne);
+  const y = Big(numTwo);
+  if (operation === '+') return x.plus(y);
+  if (operation === '-') return x.minus(y);
+  if (operation === '*') return x.times(y);
+  if (operation === '/') return x.div(y);
```
### [Optional]
It is a good practice to use `switch case` once our condition is more than 3;
kindly check out this article 👉  https://medium.com/the-daily-standup/javascript-switch-statement-vs-if-else-if-ee3111793fa0

bellom commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,13 @@
+export default function calculate(calculatorData, buttonName) {
+  let { total, next } = calculatorData;
+  const { operation } = calculatorData;
```
### [Optional]
I think we could have this destructuring in one line;
`  let { total, next, operation } = calculatorData;`

bellom commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,13 @@
+  let { total, next } = calculatorData;
+  const { operation } = calculatorData;
+
+  if (buttonName === '+/-') {
```
### Heads-Up:
**we need to add more cases;** 
1. if the `buttonName` = `A/C`, we should return `0`
2. if the `buttonName`  = `.` we should return `0.`

bellom reviewed (changes requested):
## Hi @NewIncome :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

NewIncome committed:
Make fixes from code review

- configure eslint errors locally in rules
- add big.js to calculate.js
- modify many if statements with switch-case
- make destructuring in one line
- add the cases for the buttons 'A/C' and '.'

uche-inyama reviewed (approved):
## Approved

### Hi @NewIncome  :wave: 

Go on with your next task.

### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```
